R/utilities-simulation.R

```diff
@@ -180,11 +180,22 @@ simulateModelParallel <- function(structureSets,
       pathFolder = logFolder,
       logTypes = LogTypes$Info
     )
-
-    subsetSimulationResults <- ospsuite::runSimulations(
-      simulations = simulations
-    )
-
+    
+    subsetSimulationResults <- tryCatch({
+       ospsuite::runSimulations(simulations = simulations)
+    },
+    error = function(e) {
+      logErrorThenStop(message = e, logFolder)
+      },
+    warning = function(w) {
+      logWorkflow(
+        message = w,
+        pathFolder = logFolder,
+        logTypes = LogTypes$Error
+      )
+      return(ospsuite::runSimulations(simulations = simulations))
```

Comment:
why is `ospsuite::runSimulations(simulations = simulations)` called again in the return statement?

Reply:
actually I don't quite understand why there is a return statement within tryCatch

Reply:
If some simulations worked, I wanted to still get and save them.
However, in the sub-function `warning` of `tryCatch`, I can't find a way to get the simulation results without re-calculating it

Reply:
let's return empty results then. calculating all simulations twice does not make sense for me.
The whole workflow will fail later on anyway, if some simulation results are not available.